### DISCUSSION TITLE
electron & sqlite3

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#3317] [Q&A] [answered]
now,i use imodeljs desktop with electron & sqlite3,and run success in dev IDE,
but run build:dist and run projectname.exe  but sqlite3 can not work!
who can help me!

package.json

 "sqlite3": "^5.0.2"
   "electron": "^13.1.4",
   "electron-builder": "22.10.5",

code：

  public static getdata(_dbfilepath: string, sql: string) {
        var sqlite3 = require("sqlite3").verbose();
        let db = new sqlite3.Database(_dbfilepath);
        return new Promise((resolve, reject) => {
            db.all(sql, function (err: any, data: any) {
                if (err) {
                    reject(err);
                    return undefined;
                } else {
                    return resolve(data)
                }
            })
        }
        );

    }

**Comment by pmconne** [comment, marked as answer]
Hi,

I'm not sure we can help you with sqlite3 problems - did you try contacting the maintainer of that package?
Regardless, you are going to need to be more specific than "sqlite3 can not work!"

**Reply by kabentley** [reply]
What are you using sqlite3 for? I suggest you use [SQLiteDb](https://www.itwinjs.org/reference/core-backend/sqlitedb/sqlitedb/) instead of the external "sqlite3" package.
